widget/toolbar.go

```diff
@@ -17,27 +17,63 @@ type ToolbarItem interface {
 type ToolbarAction struct {
 	Icon        fyne.Resource
 	OnActivated func() `json:"-"`
+	button      *Button
```

Comment:
You can change this field to not be a pointer. Saves you having to do a nil check. It also avoid needing to construct it manually. Just set the fields you want. Just return a reference to the field in places where it wants it to be a pointer.

Reply:
Thanks, this simplifies the code a lot.